libraries/chain/block_log.cpp

```diff
@@ -834,6 +857,56 @@ namespace eosio { namespace chain {
       }));
    }
 
+   bool prune(pruned_transaction& ptx) { 
+      ptx.prune_all();
+      return true;
+   }
+   
+   void block_log::prune_transaction(uint32_t block_num, transaction_id_type id) {
+      try {
+         EOS_ASSERT(my->version >= 4, block_log_exception, "The block log version ${version} does not support transaction pruning.", ("version", my->version));
+         uint64_t pos = get_block_pos(block_num);
+         EOS_ASSERT(pos != npos, block_log_exception,
+                     "Specified block_num ${block_num} does not exist in block log.", ("block_num", block_num));
+
+         auto entry = my->read_v4_log_entry(pos);
+
+         EOS_ASSERT(entry.block->block_num() == block_num, block_log_exception,
+                     "Wrong block was read from block log.");
+
+         auto pruner = overloaded{[](transaction_id_type&) { return false; },
+                                  [&id](pruned_transaction& ptx) { return ptx.id() == id && prune(ptx); }};
+
+         for (auto trx : entry.block->transactions) {
```

Comment:
`auto&`

Reply:
auto& doesn't work. This is an initialization statement, not an assignment statement.

Reply:
I don't follow.  auto& does work in initialization and does not work in assignment.

Reply:
Never mind, I was looking at the wrong line. I thought you were talking the auto& pruner.

Reply:
We discussed. He thought you meant line 877, not line 880.